api/src/main/java/org/apache/iceberg/actions/DeleteOrphanFiles.java

```diff
@@ -144,7 +146,25 @@ default DeleteOrphanFiles equalAuthorities(Map<String, String> newEqualAuthoriti
   @Value.Immutable
   interface Result {
     /** Returns locations of orphan files. */
-    Iterable<String> orphanFileLocations();
+    default Iterable<String> orphanFileLocations() {
+      return StreamSupport.stream(statuses().spliterator(), false)
+          .map(DeleteOrphanFiles.OrphanFileStatus::location)
+          .collect(Collectors.toList());
+    }
+
+    Iterable<OrphanFileStatus> statuses();
+  }
+
+  interface OrphanFileStatus {
```

Comment:
prefer `DeleteOrphanFileOutcome` as `FileStatus` is broadly used elsewhere, same for implementation